src/Views/KeyboardLayoutView.vala

```diff
@@ -82,36 +82,68 @@ public class KeyboardLayoutView : AbstractInstallerView {
 
         back_button.clicked.connect (() => ((Gtk.Stack) get_parent ()).visible_child = previous_view);
 
-        next_button.clicked.connect (() => next_step ());
+        next_button.clicked.connect (() => {
+            unowned Gtk.ListBoxRow row = input_variant_widget.main_listbox.get_selected_row ();
```

Comment:
I'd rather have this entire code in it's own method. It doesn't make sense to have that much functionality in a lambda. It also gets harder to debug if something goes wrong.